rust/actyx/ax/src/cmd/apps/sign.rs

```diff
@@ -16,23 +18,27 @@ pub struct SignOpts {
     pub path_to_manifest: PathBuf,
 }
 
-pub fn create_signed_app_manifest(opts: SignOpts) -> ActyxOSResult<SignedAppManifest> {
+pub fn create_signed_app_manifest(opts: SignOpts) -> ActyxOSResult<AppManifest> {
     let dev_cert = fs::read_to_string(&opts.path_to_certificate)
         .ax_err_ctx(ActyxOSCode::ERR_IO, "Failed to read developer certificate")?;
     let dev_cert: DeveloperCertificate = serde_json::from_str(&dev_cert).ax_err_ctx(
         ActyxOSCode::ERR_INVALID_INPUT,
         "Failed to deserialize developer certificate",
     )?;
+    let dev_privkey = dev_cert
+        .private_key()
+        .map(ActyxOSResult::Ok)
+        .unwrap_or_else(|| Ok(AxPrivateKey::from_file(AxPrivateKey::default_user_identity_path()?)?.to_private()))?;
     let app_manifest =
         fs::read_to_string(&opts.path_to_manifest).ax_err_ctx(ActyxOSCode::ERR_IO, "Failed to read app manifest")?;
     let app_manifest: AppManifest = serde_json::from_str(&app_manifest)
         .ax_err_ctx(ActyxOSCode::ERR_INVALID_INPUT, "Failed to deserialize app manifest")?;
 
-    let signed_manifest = SignedAppManifest::new(
-        app_manifest.app_id,
-        app_manifest.display_name,
-        app_manifest.version,
-        dev_cert.private_key(),
+    let signed_manifest = app_manifest_signer::make_signed(
```

Comment:
Why take each parameter instead of the whole `app_manifest`?

Reply:
yeah, should take an AppManifest as argument

@Kelerchian